ports/servoshell/resources.rs

```diff
@@ -47,18 +47,24 @@ fn resources_dir_path() -> PathBuf {
         path.pop();
     }
 
-    // Try ./resources in the current directory, then each of its ancestors.
-    let mut path = std::env::current_dir().unwrap();
-    loop {
-        path.push("resources");
-        if path.is_dir() {
-            *dir = Some(path);
-            return dir.clone().unwrap();
-        }
-        path.pop();
+    if cfg!(servo_production) {
+        panic!("Can't find resources directory")
+    } else {
+        // Try ./resources in the current directory, then each of its ancestors.
+        // Not to be used in production builds without considering the security implications!
+        const _: () = assert!(cfg!(servo_do_not_use_in_production));
```

Comment:
This line is causing servo build to fail locally when running `./mach build --profile production`

```
   Compiling servoshell v0.0.1 (/home/mukilan/dev/servo/ports/servoshell)
error[E0080]: evaluation of constant value failed
  --> ports/servoshell/resources.rs:55:24
   |
55 |         const _ : () = assert!(cfg!(servo_do_not_use_in_production));
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: cfg!(servo_do_not_use_in_production)', ports/servoshell/resources.rs:55:24
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not compile `servoshell` due to previous error
```

The error is gone when I remove the assignment i.e `const _:() =`. I guess the failure is due to the assertion being evaluated at compile time and it will be false for `./mach build --profile production`

Reply:
hmm, that branch should not be taken under --profile production, because we set cfg!(production). i suspect it’s because the if statement is not really conditional compilation, let me try cfg_if!{}…

Reply:
fixed in f6bb63570ddbf208d5f209421a63db1b1a5af9cb